doc/platform.rst

```diff
@@ -0,0 +1,95 @@
+
+.. class:: Platform(name, core_names=None, core_clusters=None, big_core=None, model=None, modules=None)
+
+.. method:: Platform.init_target_connection([target])
+
+    Method called by the :class:`Target` during :func:`connect`
+    This allows the platfrom to overwrite certain attributes of :class:`Target`.
+    The subclasses overwrite this method.
+
+.. method:: Platform.update_from_target([target])
+
+    Method called by the :class:`Target` during :func:`connect`.
+    This allows the :class:`Platform` to update based upon information in the :class:`Target`.
+    Overwritten by certain subclasses.
+
+.. method:: Platform.setup([target])
+
+
+    Method called by the :class:`Target` during :func:`setup`.
+    Method that can be overwritten by subclasses to provide platform-specific setup procedures.
+
+
+
+.. _gem5-platform:
+
+Gem5 Simulation Platform
+------------------------
+
+By initialising a Gem5SimulationPlatform, devlib will start a gem5 simulation (based upon the
+arguments the user provided) and then connect to it using :class:`Gem5Connection`.
+Using the methods discussed above, some methods of the :class:`Target` will be altered
+slightly to better suit gem5.
+
+.. class:: Gem5SimulationPlatform(name, host_output_dir, gem5_bin, gem5_args, gem5_virtio, gem5_telnet_port=None)
+
+    During initialisation the gem5 simulation will be kicked off (based upon the arguments
+    provided by the user) and the telnet port used by the gem5 simulation will be intercepted
+    and stored for use by the :class:`Gem5Connection`.
+
+    :param name: Platform name
+
+    :param host_output_dir: Path on the host where the gem5 outputs will be placed (e.g. stats file)
+
+    :param gem5_bin: gem5 binary
+
+    :param gem5_args: Arguments to be passed onto gem5 such as config file etc.
+
+    :param gem5_virtio: Arguments to be passed onto gem5 in terms of the virtIO device used
+                        to transfer files between the host and the gem5 simulated system.
+
+    :param gem5_telnet_port: Not yet in use as it would be used in future implementations
+                             of devlib in which the user could use the platform to pick
+                             up an existing and running simulation.
+
+
+.. method:: Platform.init_target_connection([target])
+
+    Based upon the OS defined in the :class:`Target`, the type of :class:`Gem5Connection`
+    will be set (:class:`AndroidGem5Connection` or :class:`AndroidGem5Connection`).
+
+.. method:: Platform.update_from_target([target])
+
+    This method provides specific setup procedures for a gem5 simulation. First of all, the m5
+    binary will be installed on the guest (if it is not present). Secondly, three methods
+    in the :class:`Target` will be monkey-patched:
+
+            - **reboot**: this is not supported in gem5
+            - **reset**: this is not supported in gem5
+            - **capture_screen**: gem5 might already have screencaps so the
```

Comment:
`capture_screen` doesn't seem to display properly when I build the documentation.